### ISSUE TITLE
Make UI usable by blind people

### ISSUE THREAD

**Issue by fantasai** [lukevella/rallly#306]
**Describe the bug**
One of the participants uses a screen reader and found the poll too difficult to use.

**Expected behavior**
Usable with a screenreader. :)

**Steps to Reproduce**
OK, this isn't how to actually reproduce, but it should give you an idea:
1. Replace all the the embedded SVG images with nothing.
2. Load the page in Firefox, go to View > Page Style > No Style
3. Try to use the page.

In general, if you replace all images with their alt text and show the page without CSS, it should be well-organized and readable. If it's not, it means you need to fix your markup.

**How to Improve**
1. Sorry to be throwing RTFM at you, but, you need to start by learning to use HTML tags like `<p>` and `<table>` and `<h3>` and `<section>` and `<pre>` to organize the information. Find a nice, basic introductory book or read the HTML4 spec or something. Definitely the main section of the poll should be using table markup. Here's a few relevant tutorials from MDN: [structuring text](https://developer.mozilla.org/en-US/docs/Learn/HTML/Introduction_to_HTML/HTML_text_fundamentals), [lists](https://developer.mozilla.org/en-US/docs/Learn/HTML/Introduction_to_HTML/HTML_text_fundamentals#lists), [page structure](https://developer.mozilla.org/en-US/docs/Learn/HTML/Introduction_to_HTML/Document_and_website_structure), [tables](https://developer.mozilla.org/en-US/docs/Learn/HTML/Tables/Basics).
2. Replace the embedded SVG with externally-referenced SVG using `<img>` tags, and use `alt` attributes to either mark the image as decorative (`alt=""`) or give the text that it represents (e.g. `alt="If need be"`, `alt="Total Available: "`).

There might be other things to fix as well, but this should get you 90% of the way there.

**Comment by fantasai** [comment]
This section of the CSS Writing Modes spec has some fairly complex tables, btw, you could probably just steal the markup structure: https://www.w3.org/TR/css-writing-modes-4/#unicode-bidi

**Comment by frivoal** [comment]
This issue is the main reason why I keep looking for alternatives, and cannot wholeheartedly recommend Rally.

**Comment by nesges** [comment]
This is a major issue for me as well

**Comment by hometoast** [comment]
I have a limited set of tools to validate screen reader readiness. 
For interactions with poll elements, if you tab and space to a poll option. Changing the title would trigger the reader to re-read the element?
title="Monday, September 16, 2024. No Vote"
user presses space. element's title changes.
title="Monday, September 16, 2024. Going"

**Comment by floriangouy** [comment]
Hello @fantasai and @frivoal,
A bit off topic sorry: Could you tell us which screen readers you know, so that we have examples to test and reproduce? (ie. as browers addons or standalone browsers). Thanks in advance 🙂

**Comment by lukevella** [comment]
Closing this as the original issues have been resolved. As @spthiel confirmed, the poll now uses proper table markup and the SVG icons have accessible titles.

Accessibility is an ongoing effort. We're making improvements as we refactor components rather than in one large sweep.

@spthiel if you'd like to see live regions added, feel free to open a new issue for it. Thanks everyone for the input.
